scalding-serialization-macros/src/test/scala/com/twitter/scalding/serialization/macros/MacroOrderingProperties.scala

```diff
@@ -64,8 +64,38 @@ object TestCC {
       aBB <- arb[ByteBuffer]
     } yield TestCC(aInt, aLong, anOption, aDouble, anStrOption, anOptionOfAListOfStrings, aBB)
   }
+
+  implicit def arbitraryTestCaseClassB: Arbitrary[TestCaseClassB] = Arbitrary {
+    for {
+      aInt <- arb[Int]
+      aLong <- arb[Long]
+      aDouble <- arb[Double]
+      anOption <- arb[Option[Int]]
+      anStrOption <- arb[Option[String]]
+    } yield TestCaseClassB(aInt, aLong, anOption, aDouble, anStrOption)
+  }
+
+  implicit def arbitraryTestDD: Arbitrary[TestCaseClassD] = Arbitrary {
+    for {
+      aInt <- arb[Int]
+    } yield TestCaseClassD(aInt)
+  }
+
+  implicit def arbitrarySealedTraitTest: Arbitrary[SealedTraitTest] = Arbitrary {
+    for {
+      cc <- arb[TestCC]
+      bb <- arb[TestCaseClassB]
+      dd <- arb[TestCaseClassD]
+      t <- Gen.oneOf(cc, bb, dd)
+    } yield t
+  }
 }
-case class TestCC(a: Int, b: Long, c: Option[Int], d: Double, e: Option[String], f: Option[List[String]], aBB: ByteBuffer)
+sealed trait SealedTraitTest
```

Comment:
can we deal with `case object?` For instance, I guess this macro generated code could subsume `Option` and `Either` serializers as they are just special cases of these, right?